### ISSUE TITLE
Investigate the flaky kyma-upgrade-gardener-kyma2-to-main job

### ISSUE THREAD

**Issue by VladislavPaskar** [kyma-project/kyma#13267]
**Description**

The [kyma-upgrade-gardener-kyma2-to-main](https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/kyma-upgrade-gardener-kyma2-to-main) prowjob seems to be very flaky. It is red every second time and fails with different errors like 

- the test namespace cannot be deleted
- commerceMock fixture is unready 

and so on. This job need to be investigated. 

One possible way could be adding another job, which runs this one in automated way several times in a row, so we could find a pattern.

**Comment by jakobmoellersap** [comment]
This job should now be stable again. We can close it now I think and reopen it if we see flakiness again. WDYT @VladislavPaskar ?

**Comment by jakobmoellersap** [comment]
Checked again, still tons of provisioning failures and flakiness: https://status.build.kyma-project.io/job-history/gs/kyma-prow-logs/logs/kyma-upgrade-gardener-kyma2-to-main-reconciler-main

**Comment by tobiscr** [comment]
Pipeline is read since 5 days and nobody cares about it. Sad story - do we really still need it? Seems to have no relevance for any time otherwise we would have reacted.

**Comment by tobiscr** [comment]
I would recommend to raise dedicated ticket per issue - It makes not much sense to have a "fix flaky pipeline" ticket when the issue is related to different things. Can we please close this one and start tracking the issues one by one?

**Comment by kasiakepka** [comment]
> Gardener causes provisioning timeouts (temporarily happening - retries are implemented but don't help): https://storage.googleapis.com/kyma-prow-logs/logs/kyma-upgrade-gardener-kyma2-to-main-reconciler-main/1529371773299068928/build-log.txt

Hi! See [this issue](https://github.com/kyma-project/test-infra/issues/5435), I hope it will help to solve above problem.

**Comment by arturskorupa** [comment]
Hi, please remember that according to current release schedule the last possible date to deliver code fix for the current release is 1 June in the morning. Otherwise it will have impact on the schedule.

**Comment by arturskorupa** [comment]
@k15r can you please check NATS ([issue](https://github.com/kyma-project/kyma/issues/13267#issuecomment-1137020183)
@kwiatekus please look at RAFTER
@janmedrek Commecte Mock for you ?
@strekm you already addressed ISTIO

**Comment by werdes72** [comment]
Hi, I checked failed jobs logs and Istio installs properly in all of them (or installation is not triggered at all) except one, but the problem seems to be in connection to cluster. I will keep an eye on this.

**Comment by arturskorupa** [comment]
@werdes72 then is it a blocker or not ?

**Comment by werdes72** [comment]
@arturskorupa No, I don't think so.

**Comment by arturskorupa** [comment]
I remove blocker label then. Thanks.

**Comment by arturskorupa** [comment]
I need to restore release blocker label -> there are still other parts to check:
@k15r can you please check NATS (https://github.com/kyma-project/kyma/issues/13267#issuecomment-1137020183)
@kwiatekus please look at RAFTER
@koala7659 maybe the same issue as here-> #13873

**Comment by Disper** [comment]
@arturskorupa regarding rafter failure - I would suggest to accept this aspect, as we're waiting to be unblocked with [finishing rafter removal](https://github.com/kyma-project/kyma/issues/13155).

**Comment by kwiatekus** [comment]
@arturskorupa I agree with @Disper 
@tobiscr Excluded to a separate issue
https://github.com/kyma-project/kyma/issues/14444#issuecomment-1139621703

**Comment by arturskorupa** [comment]
Current status: 
@k15r can you please check NATS (https://github.com/kyma-project/kyma/issues/13267#issuecomment-1137020183
@koala7659  Commecte Mock for you ?

**Comment by marcobebway** [comment]
Investigating this on a gardener cluster is not possible because of:

- Gardener causes provisioning timeouts https://github.com/kyma-project/kyma/issues/13267#issuecomment-1137020183

**Comment by marcobebway** [comment]
> > Gardener causes provisioning timeouts (temporarily happening - retries are implemented but don't help): https://storage.googleapis.com/kyma-prow-logs/logs/kyma-upgrade-gardener-kyma2-to-main-reconciler-main/1529371773299068928/build-log.txt
> 
> Hi! See [this issue](https://github.com/kyma-project/test-infra/issues/5435), I hope that it will help to solve above problem.

@kasiakepka when do you think the fix will be merged?

**Comment by koala7659** [comment]
I believe we should remove `release-blocker` label from that issue and leave only `failing-test`. CommerceMock tests are quite reliable at the moment after latests updates. There are some problem with external systems dependency that should be investigated as @tobiscr mentioned, but this should not stop release.

**Comment by mareknawa** [comment]
Thanks for the info @koala7659. I drop the release-blocker label then.

**Comment by mareknawa** [comment]
I've added "release blocker" back because of NATS part.
@marcobebway, I've just talked to @Halamix2 and he is checking with @kasiakepka whether they can switch to Gardener Production this flaky pipeline. They will update us soon.

**Comment by kasiakepka** [comment]
Hi, Unfortunately we are stuck on approval process and it is not possible to implement it today.

**Comment by marcobebway** [comment]
The prow job script is using two different Kyma CLI versions:

- https://github.com/kyma-project/test-infra/blob/06ea44a721d37a229c99e17fd9a325ea5ecd0012/prow/scripts/cluster-integration/kyma-upgrade-gardener-kyma2-to-main.sh#L99

- https://github.com/kyma-project/test-infra/blob/06ea44a721d37a229c99e17fd9a325ea5ecd0012/prow/scripts/cluster-integration/kyma-upgrade-gardener-kyma2-to-main.sh#L122

This was introduced in this [PR](https://github.com/kyma-project/test-infra/pull/4960) `Fixes for upgrade tests 2.0.4 -> 2.1`.

But the migration script was removed in this [PR](https://github.com/kyma-project/test-infra/pull/5172) `Remove obsolete migration script`, but still we have two different Kyma CLI versions.

@lindnerby @Tomasz-Smelcerz-SAP Do we still need to have two different Kyma CLI versions?

**Comment by mareknawa** [comment]
As agreed on PO call, I remove the "release 2.3 blocker" label from the issue. We don't have to fix it with the super urgency for 2.3 release but don't abandon the issue, please. We need to fix it because the test is unstable for long time.

**Comment by marcobebway** [comment]
# Investigation

- I ran the job multiple times to reproduce the NATS issue using this [PR](https://github.com/kyma-project/test-infra/pull/5489). 
- The failed **14** times out of **23** runs.
- I could not reproduce the NATS issue.

# Job [run](https://status.build.kyma-project.io/?job=marcobebway_test_of_prowjob_kyma-upgrade-gardener-kyma2-to-main) summary

## Provisioning errors from Gardener `[7/23]`

<details>
    <summary>More details</summary>

[gardener-provisioning-errors.log](https://github.com/kyma-project/kyma/files/8812003/gardener-provisioning-errors.log)

```
Error: All attempts fail:
#1: unable to provision gardener cluster: Provisioning timed out
```
</details>

## Fail to install Kyma components `[2/23]`

<details>
    <summary>More details</summary>

[components-install-errors-0.log](https://github.com/kyma-project/kyma/files/8812020/components-install-errors-0.log)
[components-install-errors-1.log](https://github.com/kyma-project/kyma/files/8812025/components-install-errors-1.log)

```
Deployed Components: 9/12
Could not deploy the following components:
- application-connector
- ory
- rafter
```

```
Deployed Components: 7/8
Could not deploy the following components:
- ory
```
</details>

## Fail to upgrade Kyma components `[1/23]`

<details>
    <summary>More details</summary>

[components-upgrade-errors.log](https://github.com/kyma-project/kyma/files/8812031/components-upgrade-errors.log)

```
Deployed Components: 8/12
Could not deploy the following components:
- rafter
- istio-resources
- ory
- service-catalog-addons
```
</details>

## Upgrade test errors `[4/23]`

<details>
    <summary>More details</summary>

[upgrade-test-errors-0.log](https://github.com/kyma-project/kyma/files/8812062/upgrade-test-errors-0.log)
[upgrade-test-errors-1.log](https://github.com/kyma-project/kyma/files/8812063/upgrade-test-errors-1.log)
[upgrade-test-errors-2.log](https://github.com/kyma-project/kyma/files/8812064/upgrade-test-errors-2.log)
[upgrade-test-errors-3.log](https://github.com/kyma-project/kyma/files/8812065/upgrade-test-errors-3.log)

```
18 passing (3m)
1 failing

1) Upgrade test tests
     Prometheus Tests:
       Prometheus rules should be healthy:
   AssertionError: Following rules are unhealthy: code_verb:apiserver_request_total:increase30d, code:apiserver_request_total:increase30d, code:apiserver_request_total:increase30d, cluster_verb_scope:apiserver_request_slo_duration_seconds_count:increase1h, cluster_verb_scope:apiserver_request_slo_duration_seconds_count:increase30d, cluster_verb_scope_le:apiserver_request_slo_duration_seconds_bucket:increase1h, cluster_verb_scope_le:apiserver_request_slo_duration_seconds_bucket:increase30d, apiserver_request:availability30d, apiserver_request:availability30d, apiserver_request:availability30d, code_resource:apiserver_request_total:rate5m, code_resource:apiserver_request_total:rate5m, code_verb:apiserver_request_total:increase1h, code_verb:apiserver_request_total:increase1h, code_verb:apiserver_request_total:increase1h, code_verb:apiserver_request_total:increase1h: expected [ Array(16) ] to be empty
    at Object.assertAllRulesAreHealthy (monitoring/prometheus.js:80:10)
    at async Context.<anonymous> (monitoring/index.js:52:7)
```

```
Upgrade test preparation
[ERROR] Error: Waiting for lastorder function timeout (90000 ms)
  1) CommerceMock test fixture should be ready
  ✓ Helm Broker test fixture should be ready (36211ms)


1 passing (2m)
1 failing

1) Upgrade test preparation
     CommerceMock test fixture should be ready:
   HttpError: HTTP request failed
    at Request._callback (node_modules/@kubernetes/client-node/dist/object.js:416:32)
    at Request.self.callback (node_modules/request/request.js:185:22)
    at Request.<anonymous> (node_modules/request/request.js:1154:10)
    at IncomingMessage.<anonymous> (node_modules/request/request.js:1076:12)
    at endReadableNT (internal/streams/readable.js:1333:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
```

```
Upgrade test preparation
[ERROR] Error: Waiting for lastorder function timeout (90000 ms)
  1) CommerceMock test fixture should be ready
[ERROR] Error: Waiting for testing service class timeout (90000 ms)
  2) Helm Broker test fixture should be ready


0 passing (5m)
2 failing

1) Upgrade test preparation
     CommerceMock test fixture should be ready:
   HttpError: HTTP request failed
    at Request._callback (node_modules/@kubernetes/client-node/dist/object.js:416:32)
    at Request.self.callback (node_modules/request/request.js:185:22)
    at Request.<anonymous> (node_modules/request/request.js:1154:10)
    at IncomingMessage.<anonymous> (node_modules/request/request.js:1076:12)
    at endReadableNT (internal/streams/readable.js:1333:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)

2) Upgrade test preparation
     Helm Broker test fixture should be ready:
   Error: Waiting for testing service class timeout (90000 ms)
    at Timeout._onTimeout (utils/index.js:395:20)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
```

```
Upgrade test preparation
[ERROR] Error: Waiting for lastorder function timeout (90000 ms)
  1) CommerceMock test fixture should be ready
[ERROR] Error: Waiting for testing service class timeout (90000 ms)
  2) Helm Broker test fixture should be ready


0 passing (6m)
2 failing

1) Upgrade test preparation
     CommerceMock test fixture should be ready:
   Error: Waiting for lastorder function timeout (90000 ms)
    at Timeout._onTimeout (utils/index.js:395:20)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)

2) Upgrade test preparation
     Helm Broker test fixture should be ready:
   Error: Waiting for testing service class timeout (90000 ms)
    at Timeout._onTimeout (utils/index.js:395:20)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
```
</details>

## Successful runs `[9/23]`

**Comment by marcobebway** [comment]
# Investigation

- I ran the job multiple times to reproduce the NATS issue using this [PR](https://github.com/kyma-project/test-infra/pull/5489). 
- The job failed **14** times out of **23** runs.
- I could not reproduce the NATS issue.

# Job [run](https://status.build.kyma-project.io/?job=marcobebway_test_of_prowjob_kyma-upgrade-gardener-kyma2-to-main) summary

## Provisioning errors from Gardener `[7/23]`

<details>
    <summary>More details</summary>

[gardener-provisioning-errors.log](https://github.com/kyma-project/kyma/files/8812003/gardener-provisioning-errors.log)

```
Error: All attempts fail:
#1: unable to provision gardener cluster: Provisioning timed out
```
</details>

## Fail to install Kyma components `[2/23]`

<details>
    <summary>More details</summary>

[components-install-errors-0.log](https://github.com/kyma-project/kyma/files/8812020/components-install-errors-0.log)
[components-install-errors-1.log](https://github.com/kyma-project/kyma/files/8812025/components-install-errors-1.log)

```
Deployed Components: 9/12
Could not deploy the following components:
- application-connector
- ory
- rafter
```

```
Deployed Components: 7/8
Could not deploy the following components:
- ory
```
</details>

## Fail to upgrade Kyma components `[1/23]`

<details>
    <summary>More details</summary>

[components-upgrade-errors.log](https://github.com/kyma-project/kyma/files/8812031/components-upgrade-errors.log)

```
Deployed Components: 8/12
Could not deploy the following components:
- rafter
- istio-resources
- ory
- service-catalog-addons
```
</details>

## Upgrade test errors `[4/23]`

<details>
    <summary>More details</summary>

[upgrade-test-errors-0.log](https://github.com/kyma-project/kyma/files/8812062/upgrade-test-errors-0.log)
[upgrade-test-errors-1.log](https://github.com/kyma-project/kyma/files/8812063/upgrade-test-errors-1.log)
[upgrade-test-errors-2.log](https://github.com/kyma-project/kyma/files/8812064/upgrade-test-errors-2.log)
[upgrade-test-errors-3.log](https://github.com/kyma-project/kyma/files/8812065/upgrade-test-errors-3.log)

```
18 passing (3m)
1 failing

1) Upgrade test tests
     Prometheus Tests:
       Prometheus rules should be healthy:
   AssertionError: Following rules are unhealthy: code_verb:apiserver_request_total:increase30d, code:apiserver_request_total:increase30d, code:apiserver_request_total:increase30d, cluster_verb_scope:apiserver_request_slo_duration_seconds_count:increase1h, cluster_verb_scope:apiserver_request_slo_duration_seconds_count:increase30d, cluster_verb_scope_le:apiserver_request_slo_duration_seconds_bucket:increase1h, cluster_verb_scope_le:apiserver_request_slo_duration_seconds_bucket:increase30d, apiserver_request:availability30d, apiserver_request:availability30d, apiserver_request:availability30d, code_resource:apiserver_request_total:rate5m, code_resource:apiserver_request_total:rate5m, code_verb:apiserver_request_total:increase1h, code_verb:apiserver_request_total:increase1h, code_verb:apiserver_request_total:increase1h, code_verb:apiserver_request_total:increase1h: expected [ Array(16) ] to be empty
    at Object.assertAllRulesAreHealthy (monitoring/prometheus.js:80:10)
    at async Context.<anonymous> (monitoring/index.js:52:7)
```

```
Upgrade test preparation
[ERROR] Error: Waiting for lastorder function timeout (90000 ms)
  1) CommerceMock test fixture should be ready
  ✓ Helm Broker test fixture should be ready (36211ms)


1 passing (2m)
1 failing

1) Upgrade test preparation
     CommerceMock test fixture should be ready:
   HttpError: HTTP request failed
    at Request._callback (node_modules/@kubernetes/client-node/dist/object.js:416:32)
    at Request.self.callback (node_modules/request/request.js:185:22)
    at Request.<anonymous> (node_modules/request/request.js:1154:10)
    at IncomingMessage.<anonymous> (node_modules/request/request.js:1076:12)
    at endReadableNT (internal/streams/readable.js:1333:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)
```

```
Upgrade test preparation
[ERROR] Error: Waiting for lastorder function timeout (90000 ms)
  1) CommerceMock test fixture should be ready
[ERROR] Error: Waiting for testing service class timeout (90000 ms)
  2) Helm Broker test fixture should be ready


0 passing (5m)
2 failing

1) Upgrade test preparation
     CommerceMock test fixture should be ready:
   HttpError: HTTP request failed
    at Request._callback (node_modules/@kubernetes/client-node/dist/object.js:416:32)
    at Request.self.callback (node_modules/request/request.js:185:22)
    at Request.<anonymous> (node_modules/request/request.js:1154:10)
    at IncomingMessage.<anonymous> (node_modules/request/request.js:1076:12)
    at endReadableNT (internal/streams/readable.js:1333:12)
    at processTicksAndRejections (internal/process/task_queues.js:82:21)

2) Upgrade test preparation
     Helm Broker test fixture should be ready:
   Error: Waiting for testing service class timeout (90000 ms)
    at Timeout._onTimeout (utils/index.js:395:20)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
```

```
Upgrade test preparation
[ERROR] Error: Waiting for lastorder function timeout (90000 ms)
  1) CommerceMock test fixture should be ready
[ERROR] Error: Waiting for testing service class timeout (90000 ms)
  2) Helm Broker test fixture should be ready


0 passing (6m)
2 failing

1) Upgrade test preparation
     CommerceMock test fixture should be ready:
   Error: Waiting for lastorder function timeout (90000 ms)
    at Timeout._onTimeout (utils/index.js:395:20)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)

2) Upgrade test preparation
     Helm Broker test fixture should be ready:
   Error: Waiting for testing service class timeout (90000 ms)
    at Timeout._onTimeout (utils/index.js:395:20)
    at listOnTimeout (internal/timers.js:557:17)
    at processTimers (internal/timers.js:500:7)
```
</details>

## Successful runs `[9/23]`
